src/main/java/org/broadinstitute/hellbender/tools/spark/BaseRecalibratorSpark.java

```diff
@@ -0,0 +1,116 @@
+
+import java.io.PrintStream;
+import java.util.List;
+
+@CommandLineProgramProperties(
+        summary = "Base Quality Score Recalibration (BQSR) -- Generates recalibration table based on various user-specified covariates (such as read group, reported quality score, machine cycle, and nucleotide context).",
+        oneLineSummary = "Generates recalibration table",
+        programGroup = ReadProgramGroup.class
+)
+public class BaseRecalibratorSpark extends SparkCommandLineProgram {
+    private static final long serialVersionUID = 1L;
+
+    /**
+     * all the command line arguments for BQSR and its covariates
+     */
+    @ArgumentCollection(doc = "all the command line arguments for BQSR and its covariates")
+    private final RecalibrationArgumentCollection bqsrArgs = new RecalibrationArgumentCollection();
+
+    @ArgumentCollection
+    private final RequiredReadInputArgumentCollection readArguments = new RequiredReadInputArgumentCollection();
+
+    @ArgumentCollection
+    private final IntervalArgumentCollection intervalArgumentCollection = new OptionalIntervalArgumentCollection();
+
+    @ArgumentCollection
+    private final ReferenceInputArgumentCollection referenceArguments = new OptionalReferenceInputArgumentCollection();
+
+    @Argument(doc = "the known variants", shortName = "knownSites", fullName = "knownSites", optional = false)
+    private List<String> knownVariants;
+
+    @Argument(doc = "Path to save the final recalibration tables to.",
+              shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME, fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME, optional = false)
+    private String outputTablesPath = null;
+
+    @Override
+    protected void runPipeline( JavaSparkContext ctx ) {
+        if ( readArguments.getReadFilesNames().size() != 1 ) {
+            throw new UserException("Sorry, we only support a single reads input for now.");
+        }
+        final String bam = readArguments.getReadFilesNames().get(0);
+
+        ReadsSparkSource readSource = new ReadsSparkSource(ctx);
+        SAMFileHeader readsHeader = ReadsSparkSource.getHeader(ctx, bam);
+        final List<SimpleInterval> intervals = intervalArgumentCollection.intervalsSpecified() ? intervalArgumentCollection.getIntervals(readsHeader.getSequenceDictionary())
+                : IntervalUtils.getAllIntervalsForReference(readsHeader.getSequenceDictionary());
+        JavaRDD<GATKRead> initialReads = readSource.getParallelReads(bam, intervals);
+
+        VariantsSparkSource variantsSparkSource = new VariantsSparkSource(ctx);
+        if ( knownVariants.size() > 1 ) {
+            throw new GATKException("Cannot currently handle more than one known sites file, " +
+                    "as getParallelVariants(List) is broken");
+        }
+        JavaRDD<Variant> bqsrKnownVariants = variantsSparkSource.getParallelVariants(knownVariants.get(0));
+
+        GCSOptions options = BucketUtils.getAuthenticatedGCSOptions(apiKey);
```

Comment:
It should be possible to read reference data from the file system too (a FASTA file from the local filesystem or HDFS), see https://github.com/broadinstitute/hellbender/blob/master/src/main/java/org/broadinstitute/hellbender/engine/dataflow/datasources/ReferenceDataflowSource.java#L37.

This simplest fix here would be to check if `apiKey` is null, and if it is set `options` to null.